librz/core/visual.c

```diff
@@ -2522,13 +2522,17 @@ RZ_API int rz_core_visual_cmd(RzCore *core, const char *arg) {
 		case 'c':
 			setcursor(core, !core->print->cur_enabled);
 			break;
-		case '$':
-			if (core->print->cur_enabled) {
```

Comment:
I would convert all these `dr <reg>=<val>` a bit differently. I think the command handler for `dr` does slightly different things (e.g. it consider whether the reg is flags or not), so I would refactor a bit that part and use a function there instead of doing this. See how I did it for `ar <reg>=<val>` with `rz_core_analysis_set_reg`.

Reply:
I think it's not required if we know the reg is PC which is the case here.

Reply:
I know, but I still think it's better to have the same code here, for consistency. In particular, the `dr <reg>=<val>` code should be refactored in a function similar to `rz_core_analysis_set_reg` that takes a `regname` and a `val` as arguments. For example, `dr` also does `rz_core_debug_regs2flags` which is not done here. Please introduce that API now, because otherwise later it will become almost impossible to understand that these pieces of code were actually using `dr`.

Reply:
Ok, done. I introduced new `RZ_IPI bool rz_core_debug_reg_set(RzCore *core, const char *regname, ut64 regval, const char *ostr)`. Note the two arguments, I know it's a bit ugly but I have to use both because of how `dr` is actually implemented. Let me know if you have better ideas.